sys/Makefile

```diff
@@ -12,13 +12,22 @@ CLEAN-FILES += mimiker.elf.map
 
 include $(TOPDIR)/build/build.kern.mk
 
-LDFLAGS += -T $(TOPDIR)/sys/$(ARCH)/$(ARCH).ld
+LDFLAGS += -T $(TOPDIR)/sys/kernel.ld
 LDLIBS	= -Wl,--start-group \
 	    -Wl,--whole-archive $(KLIBLIST) -Wl,--no-whole-archive \
             -lgcc \
           -Wl,--end-group
 
-mimiker.elf: $(KLIBDEPS)
+kernel.ld: $(ARCH)/$(ARCH).ld
+	@echo "[CPP] $(DIR)$< -> $(DIR)$@"
+ifeq ($(CPPLDSCRIPT), 1)
+	$(CPP) $(CPPFLAGS) -I$(TOPDIR)/include $< | grep -v '^#' | \
+		grep -v '^$$' > $@
+else
+	$(CP) $< $@
+endif
```

Comment:
Please run *man gcc* and read it *carefully*.
You can remove this error prone magic and use
`$(CPP) $(CPPFLAGS) -I$(TOPDIR)/include -P -o $@ $<`
I also don't understand why do you need `CPPLDSCRIPT` - I removed that if statement and it works for AArch64.
I see that there is a problem with mips but you can fixed it by force undef & define (and it's a common solution for that).

Reply:
I took the formula directly form NetBSD, but I agree your solution is much better! Regarding conditional CPP processing (i.e. `CPPLDSCRIPT`), I agree it's a weird idea and it also has been removed. @cahirwpz proposed introducing optional `$(ARCH).ld.in` files generating `$ARCH).ld` files.

Reply:
Looks much better now.